### PR TITLE
Development

### PR DESCRIPTION
**Pull request by ameliachoo**

### CODE REVIEW

**Comment by RageCatalyst** [comment]
## Feedback
### Style
- Class names match file names
- File names in Pascal Case
- Space after for/if/while
- Braces always used

### Requirements
- All requirements met
	- Even additional functionality

### Readability
- The code is easy to read thanks to detailed comments
	- XML Comments present

### Error Handling
- No error handling present
	- However program will never crash since an empty string is still accepted.
	- Other inputs will still be accepted.
## Critiques
### Style
- Indentations are tabs, not double spaces
- Private variables $\neq$ `_camelCase
- Local/public variables in "Die" class are in PascalCase
- Line break before opening brace
- Line break after close brace and else

### Testing
- More tests could be added

File: CMP1903_A1_2324/Die.cs
Comment (SChilton1712) [inline CMP1903_A1_2324/Die.cs:1]: Initialisation of DiceRoll with a random value is unnecessary, as the only function in that class already assigns a value to DiceRoll before it is returned.

File: CMP1903_A1_2324/Testing.cs
Comment (SChilton1712) [inline CMP1903_A1_2324/Testing.cs:1]: The Console.WriteLine() at the end of the method runs regardless of whether the program is in Debug mode or Release mode - it runs even if the tests don't. Consider using Debug.WriteLine() instead.

**Review by SChilton1712 — COMMENTED** [review]
### Code Review

**Specifications:**
- All required specifications met.
- Optional specifications also met.

**Practices**
- Comments are frequent, and are applied effectively.
  - Classes are headed with XML-compatible comments.
- Full adherence to the style guide, accounting for:
  - Variable identifiers.
  - Class names and file names.
  - Application of braces.
  - Et cetera.

**Suggestions**

- (Die class, line 17): Initialisation of the DiceRoll variable of the Die class with a call to _rnd is unnecessary, as it is only accessed via the Roll() method, which sets the value anyway.
- (Testing class, line 18): The RunTest() method only has a single return statement, and only returns 0. Consider utilising a void method instead, as it doesn't need a return statment.
- (Testing class, line 38): Console.WriteLine() runs regardless of if the program is in Release mode or Debug mode, so that line runs even if the tests don't. For this reason, consider using Debug.WriteLine(), which functions the same but only in Debug mode.

**Pull request merged**